mobile/src/main/java/org/openhab/habdroid/util/ExtensionFuncs.kt

```diff
@@ -177,6 +177,11 @@ inline fun <T> JSONArray.map(transform: (JSONObject) -> T): List<T> {
     return (0 until length()).map { index -> transform(getJSONObject(index)) }
 }
 
+fun JSONObject.getStringOrNull(key: String): String? {
+    val value = optString(key, "")
```

Comment:
Should probably implement this as `return if (has(key)) getString(key) else null` since otherwise a field which is present but empty is treated like a missing field.

Reply:
Done.